### PR TITLE
Support glob pattern in skip paths

### LINKED ISSUE #64: Feature request: support globs for skipped files
Currently you have to specify plain paths for files you want to skip when updating the cruft repository. It would be nice if it was possible to use globs in skipped paths like so:

```json
    {
        "skip": [
            "**/__init__.py",
            "tests/**"
        ]
    }
```

By the way, ideally, the cookiecutter template would even define some of those skipped files.

If this sounds acceptable to you, I'm inclined to contributing such feature once #59 makes it to master (otherwise it'd conflict).

### PR DESCRIPTION
**Pull request by daavoo**
Closes #64 
This P.R. allows to use glob patterns for selecting the files to skip.

### CODE REVIEW

File: cruft/_commands/utils/generate.py [hunk 1]
```diff
@@ -112,10 +112,10 @@ def _get_deleted_files(template_dir: Path, project_dir: Path):
...
+def _remove_paths(root: Path, paths_to_remove: Union[Set[Path], Set[str]]):
     for path_to_remove in paths_to_remove:
-        path = root / path_to_remove
-        if path.is_dir():
-            rmtree(path)
-        elif path.is_file():
-            path.unlink()
+        for path in root.glob(str(path_to_remove)):
+            if path.is_dir():
+                rmtree(path)
```
Comment (samj1912) [inline cruft/_commands/utils/generate.py:119]: @daavoo can you please add tests so that the missing lines are covered?

File: tests/test_api.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def test_update_and_check_real_repo(tmpdir):
     repo.head.reset(commit="86a6e6beda8095690414ff7652c15b7ae36e6128", working_tree=True)
     with open(os.path.join(tmpdir, ".cruft.json")) as cruft_file:
         cruft_state = json.load(cruft_file)
-        cruft_state["skip"] = ["cruft/__init__.py", "tests"]
+        cruft_state["skip"] = ["cruft/__init__.py", "tests/*"]
```
Comment (samj1912) [inline tests/test_api.py:80]: Is this changed needed? Are top level folders not supported any more?

**Review by samj1912 — CHANGES_REQUESTED** [review]

File: cruft/_commands/utils/generate.py (hunk 1 above)
Comment (daavoo) [reply cruft/_commands/utils/generate.py:119]: done

File: tests/test_api.py (hunk 2 above)
Comment (daavoo) [reply tests/test_api.py]: The change is not needed, in fact the behavior would be the same in the 2 cases. I should had not pushed that change, I was debugging some variations locally.

**Review by samj1912 — APPROVED** [review]

**Pull request merged**